pkg/system/system.go

```
@@ -167,8 +169,27 @@ func (sc *Controller) registerRouter() {
 	sc.router.HandleFunc(endpointDaemons, sc.describeDaemons()).Methods(http.MethodGet)
 	sc.router.HandleFunc(endpointDaemonsUpgrade, sc.upgradeDaemons()).Methods(http.MethodPut)
 	sc.router.HandleFunc(endpointDaemonRecords, sc.getDaemonRecords()).Methods(http.MethodGet)
+	sc.router.HandleFunc(endpointPrefetch, sc.setPrefetchFiles()).Methods(http.MethodPut)
 }
 
+func (sc *Controller) setPrefetchFiles() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			log.L.Errorf("Failed to read prefetch list: %v", err)
+			http.Error(w, "Failed to read prefetch list", http.StatusInternalServerError)
```

Should we return this error to client?